CONTRIBUTION.md

```diff
@@ -33,7 +34,223 @@ These commands can be executed from the project root:
+				return nil
+         }
+			return outputResult(cmd, model.OutputFormat, instances)
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+// Configure command flags (type, default value, and description)
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().StringP(myFlag, "defaultValue", "My flag description")
+}
+
+// Parse user input (arguments and/or flags)
+func parseInput(cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+   myArg := inputArgs[0]
+
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	return &inputModel{
+		GlobalFlagModel: globalFlags,
+      MyArg            myArg,
+		MyFlag:          flags.FlagToStringPointer(cmd, myFlag),
+	}, nil
+}
+
+// Build request to the API
+func buildRequest(ctx context.Context, model *inputModel, apiClient *foo.APIClient) (foo.ApiListInstancesRequest error) {
+   someParam, err := utils.GetParam(model.MyFlag) // Util function to get API parameter based on flag value
+   if err != nil {
+      return fmt.Errorf("(...): %w", err)
+   }
+	req := apiClient.GetBar(ctx, model.ProjectId, model.MyArg, someParam)
+	return req
+}
+
+// Output result based on the configured output format
+func outputResult(cmd *cobra.Command, outputFormat string, resources []foo.Resource) error {
+	switch outputFormat {
+	case globalflags.JSONOutputFormat:
+		details, err := json.MarshalIndent(instances, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal resource list: %w", err)
+		}
+		p.Outputln(string(details))
+		return nil
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "NAME", "STATE")
+		for i := range resources {
+			resource := resources[i]
+			table.AddRow(*resource.ResourceId, *resource.Name, *resource.State)
+		}
+		err := table.Display(cmd)
+		if err != nil {
+			return fmt.Errorf("render table: %w", err)
+		}
+		return nil
+	}
+}
+```
+
+Please remeber to always add unit tests for `parseInput`, `buildRequest` (in `bar_test.go`), and any other util functions used.
+
+If the new command `bar` is the first command in the CLI using a STACKIT service `foo`, please refer to [Onboarding a new STACKIT service](./CONTRIBUTION.md/#onboarding-a-new-stackit-service).
+
+#### Outputs, prints and debug logs
+
+The CLI has 4 different verbosity levels:
+
+- `error`: For only displaying errors
+- `warning`: For displaying user facing warnings _(and all of the above)_
+- `info` (default): For displaying user facing info, such as operation success messages and spinners _(and all of the above)_
+- `debug`: For displaying structured logs with different levels, including errors _(and all of the above)_
+
+For prints that are specific to a certain log level, you can use the methods defined in the `print` package: `Error`, `Warn`, `Info`, and `Debug`.
+
+For command outputs that should always be displayed, no matter the defined verbosity, you should use the `print` methods `Outputf` and `Outputln`.
```

Comment:
Make it more clear that these methods should only be used for the command output, which are the outputs described by "I run the command to see this"